src/python/sansmic/io.py

```diff
@@ -326,12 +331,25 @@ def read_dat(str_or_buffer: Union[str, Path], *, ignore_errors=False) -> Scenari
                     )
                 )
                 raise TypeError("Invalid data in DAT file: RESETGEO not 0")
-            if not first and isinstance(cavern_sg, (float, int)) and cavern_sg > 1.0:
+            if float(tDelay) != 0:
+                logger.warning("The TDLAY option should not be used. Ignoring.")
+                scenario.comments += "* The TDLAY option ({}) was removed from stage {}, as it has been deprecated.\n".format(
+                    tDelay, len(scenario.stages) + 1
+                )
+            if not first and (cavern_sg is not None and float(cavern_sg) > 1.0):
+                scenario.comments += "* Initial cavern SG was changed from {} to 0 in subsequent stage number {}.\n".format(
+                    cavern_sg, len(scenario.stages) + 1
+                )
+                scenario.comments += "  If you truly meant to reset the cavern SG at the beginning of the stage,\n"
+                scenario.comments += "  please add ``set-cavern-sg = {}`` into [[stages]] definition number {}\n".format(
+                    cavern_sg, len(scenario.stages) + 1
+                )
                 logger.warning(
-                    "The REPEAT option was supposed to turn off the cavern SG; it did not do so. sansmic is currently mimicing SANSMIC behavior and resetting the cavern brine to {} sg in stage {}. \n\nIf this is not what is intended, please manually remove the 'set-cavern-sg' entry from stages after stage 1. This behavior will change in future releases.".format(
+                    "The REPEAT option now disables cavern SG initialization that is non-zero. Please add ``set-initial-conditions = true`` and ``set-cavern-sg = {}`` to [[stages]] {}".format(
                         cavern_sg, len(scenario.stages) + 1
                     )
                 )
+                cavern_sg = 0.0
             stage.title = title
```

Comment:
@hgmaure the other changes are enhancements/modifications to logging the conversion process. The above change is where initial cavern SG is now always changed to 0.0 (continue from last stage) if a stage is a repeat/subsequent stage. The undocumented, prior behavior can still be forced to occur if both ``set-initial-conditions = true`` and ``set-cavern-sg = 1.2019`` are added to a stage definition by hand.

Reply:
Okay, more than that line. I'm not sure why it only added it to that line.